modules/ui/sections/map_daterange.js

```diff
@@ -0,0 +1,140 @@
+        .label(() => t.append('date_ranges.title'))
+        .disclosureContent(renderDisclosureContent)
+        .expandedByDefault(false);
+
+    function renderDisclosureContent(selection) {
+        const container = selection.selectAll('.date_ranges-container').data([0]);
+
+        // start date label & input
+        const mindate_label = container.enter()
+            .append('label')
+            .html(t('date_ranges.start_date.description'))
+            .merge(container);
+        const mindate_input = container.enter()
+            .append('input')
+            .attr('type', 'text')
+            .attr('value', DEFAULT_MIN_DATE)
+            .attr('title', () => t.append('date_ranges.start_date.tooltip'))
+            .attr('placeholder', () => t.append('date_ranges.start_date.placeholder'))
+            .merge(container);
+
+        // line break
+        container.enter()
+            .append('br')
+            .merge(container);
+
+        // end date label & input
+        const maxdate_label = container.enter()
+            .append('label')
+            .html(t('date_ranges.end_date.description'))
+            .merge(container);
+        const maxdate_input = container.enter()  // we will refer to this widget by its name attribute to fetch our date range
+            .append('input')
+            .attr('type', 'text')
+            .attr('value', DEFAULT_MAX_DATE)
+            .attr('title', () => t.append('date_ranges.end_date.tooltip'))
+            .attr('placeholder', () => t.append('date_ranges.end_date.placeholder'))
+            .merge(container);
+
+        // apply styles
+        INPUT_STYLES.forEach(function (style) {
+            mindate_input.style(style.name, style.value);
+            maxdate_input.style(style.name, style.value);
+        });
+        LABEL_STYLES.forEach(function (style) {
+            mindate_label.style(style.name, style.value);
+            maxdate_label.style(style.name, style.value);
+        });
+
+        // event handler for change event
+        // intercept invalid & blank and correct them to our hardcoded in/max
+        // then cause a re-filter/redraw
+        function applyDateRange() {
+            const mindate = mindate_input.property('value');
+            const maxdate = maxdate_input.property('value');
+
+            context.features().dateRange = [mindate, maxdate];
+            context.features().redraw();
+
+            updateUrlParam();
+        }
+
+        function ensureValidInputs() {
+            // if utilNormalizeDateString() can make sense of it, so can "date_range" in renderer/features.js
+            // if not, then complain and reset to the starting value
+            const mindate = mindate_input.property('value');
+            const maxdate = maxdate_input.property('value');
+
+            if (!utilNormalizeDateString(mindate)) {
+                mindate_input.property('value', DEFAULT_MIN_DATE);
+            }
```

Comment:
If the user enters a malformed but guessable date, this PR currently relies on `featureFitsDateRange()` to convert it to that guess behind the scenes. This is fine for `5 October 2022`, but for something like `5/10/2022`, it’s unclear to the user whether it interpreted the input correctly. If we unconditionally replace the `value` with `utilNormalizeDateString()`’s `value`, then it becomes more transparent to the user.

Incidentally, this would actually ensure valid inputs, so that `featureFitsDateRange()` won’t have to perform the same normalization every time it evaluates a feature.

Reply:
> we unconditionally replace the value with utilNormalizeDateString()’s value, then it becomes more transparent to the user

I gotta say, the visual effect here is nice. I enter a date, and it is visibly changed and displayed in front of me. This was a good idea.